### PR TITLE
Avoid using FQN

### PR DESCRIPTION
**Pull request by lucasmichot**
As this is already the case where most FQN are imported, this PR skip the usage of FQN in the source

### CODE REVIEW

**Comment by gordalina** [comment]
:+1:

File: build [hunk 1]
```diff
@@ -1,5 +1,8 @@
 #!/usr/bin/env php
 <?php
+use Phar;
```
Comment (GrahamCampbell) [inline build]: missing newline

File: src/Deployer.php [hunk 2]
```diff
@@ -134,14 +140,14 @@ public function getOutput()
     /**
      * @param string $name
      * @return mixed
-     * @throws \InvalidArgumentException
+     * @throws InvalidArgumentException
```
Comment (GrahamCampbell) [inline src/Deployer.php]: :-1:

Comment (GrahamCampbell) [inline src/Deployer.php]: FQCNs should always be used in phpdoc IMO.

File: build (hunk 1 above)
Comment (lucasmichot) [inline build]: Is it PSR2 ?

Comment (GrahamCampbell) [inline build]: no, though the rest of the code base does that...

File: src/Deployer.php (hunk 2 above)
Comment (gordalina) [inline src/Deployer.php]: I disagree. FQCNs should be used if class is not part of file's namespace IMO.
So here it would be `\InvalidArgumentException`

Comment (lucasmichot) [inline src/Deployer.php]: I am preparing a PR regarding the usage of FQN in docblocks

Comment (GrahamCampbell) [inline src/Deployer.php]: @gordalina I don't understand why you'd want to make phpdoc hard to read? The whole point of phpdoc is for people to read, not for silly IDEs.

File: build (hunk 1 above)
Comment (lucasmichot) [inline build]: Would make sense to add some more PHP-CS-Fixer fixers, but it's a bit on @elfet side

File: src/Deployer.php (hunk 2 above)
Comment (gordalina) [inline src/Deployer.php]: @GrahamCampbell i'm advocating the use of class names as you'd do if you were writing code. If you `use` a class or a namespace, then you should use that in the doc-block as well. 
Regarding your comment, i'm making it the same to read, you are making it hard to write.

**Comment by elfet** [comment]
Again: why we need this?

**Comment by lucasmichot** [comment]
It allows to have a more concise code.
Out of the function the developer has no clue of the arguments, so why the docblocks FQN are useful
But within the function it does not really makes sense to repeat many times the same namespace.

Also in the current code, sometimes the namespace is imported, sometimes it is not.

Importing FQN in code provide a common practice for all developer who wish to contribute to the project.
ATM, they don't have any clue what to do.

**Comment by GrahamCampbell** [comment]
:+1:

File: bin/dep [hunk 3]
```diff
@@ -34,9 +38,9 @@ require_once 'src/functions.php';
 define('DEPLOYER_BIN', __FILE__);
 
 // Init Deployer
-$console = new \Deployer\Console\Application('Deployer', 'master');
-$input = new \Symfony\Component\Console\Input\ArgvInput();
-$output = new \Symfony\Component\Console\Output\ConsoleOutput();
+$console = new Application('Deployer', 'master');
+$input = new ArgvInput();
+$output = new ConsoleOutput();
 $deployer = new \Deployer\Deployer($console, $input, $output);
```
Comment (gordalina) [inline bin/dep]: Why not use `Deployer\Deployer` here?

Comment (lucasmichot) [inline bin/dep]: I've skipped this one, my bad :+1:

**Comment by lucasmichot** [comment]
BTW @elfet : https://github.com/deployphp/deployer/pull/368#discussion_r32720683
Some framework use lots of fixers https://github.com/laravel/framework/blob/5.1/.php_cs
Some less, it's really up to you

**Comment by gordalina** [comment]
I would vote for a cs fixer, see why: https://github.com/deployphp/deployer/pull/370#issuecomment-113318748 (3rd paragraph)

**Comment by lucasmichot** [comment]
Sure @gordalina and it's already implemented : https://github.com/deployphp/deployer/blob/master/.php_cs :-)

But shall we add more fixers to this .php_cs definition file ?

**Comment by gordalina** [comment]
isn't there any CI that validates a PR with cs fixer? 
that would force correctness for all new code :smile:

**Comment by GrahamCampbell** [comment]
> isn't there any CI that validates a PR with cs fixer? 

Yeh. https://styleci.io. :)

**Comment by GrahamCampbell** [comment]
If you like the rules laravel uses, you can use the "laravel" preset on styleci. :)

**Comment by GrahamCampbell** [comment]
https://github.com/StyleCI/Config/blob/master/src/Config.php

**Comment by gordalina** [comment]
@GrahamCampbell does it notify the PR that its valid/invalid?
For example, this PR has no info from StyleCI

**Comment by GrahamCampbell** [comment]
This PR has no info if it was sent after StyleCI was enabled. I think open/closeing this PR will trigger StyleCI to run. :)

**Comment by GrahamCampbell** [comment]
> does it notify the PR that its valid/invalid?

Yes it does. It uses github's status API.

**Comment by gordalina** [comment]
Awesome then!
Then ignore all my comments, as everything was setup before i've opened my mouth.

**Comment by lucasmichot** [comment]
> This PR has no info if it was sent before StyleCI was enabled. I think open/closeing this PR will trigger StyleCI to run. :)

Can anyone @gordalina @elfet @ZeeCoder try to close and re-open this PR ?

**Comment by GrahamCampbell** [comment]
@lucasmichot Can't you do it?

**Comment by GrahamCampbell** [comment]
> I think open/closeing this PR will 

Looking at my code, the 3rd way of doing it is just to push to this PR. :)

**Comment by GrahamCampbell** [comment]
A rebase and force push would do it for example.

**Comment by lucasmichot** [comment]
Rebase or forced push

**Pull request closed without merging**